matplotlib.docstring is deprecated

Warnings are being emitted from matplotlib that will lead to an incompatibility very soon [[ref](https://github.com/gwpy/gwpy/actions/runs/3412192448/jobs/5677377037)]:

```pytb
gwpy/__init__.py:34: in <module>
    from . import (
gwpy/plot/__init__.py:29: in <module>
    from . import (
gwpy/plot/gps.py:28: in <module>
    from matplotlib import (ticker, docstring)
../../../miniconda3/envs/test/lib/python3.8/site-packages/matplotlib/docstring.py:3: in <module>
    _api.warn_deprecated(
../../../miniconda3/envs/test/lib/python3.8/site-packages/matplotlib/_api/deprecation.py:96: in warn_deprecated
    warn_external(warning, category=MatplotlibDeprecationWarning)
../../../miniconda3/envs/test/lib/python3.8/site-packages/matplotlib/_api/__init__.py:363: in warn_external
    warnings.warn(message, category, stacklevel)
E   matplotlib._api.deprecation.MatplotlibDeprecationWarning: The matplotlib.docstring module was deprecated in Matplotlib 3.6 and will be removed two minor releases later.
```

--- a/gwpy/plot/gps.py
+++ b/gwpy/plot/gps.py
@@ -25,13 +25,21 @@ from numbers import Number
 
 import numpy
 
-from matplotlib import (ticker, docstring)
+from matplotlib import ticker
 from matplotlib.scale import (register_scale, LinearScale, get_scale_names)
 from matplotlib.transforms import Transform
 try:
     from matplotlib.scale import _get_scale_docs as get_scale_docs
 except ImportError:  # matplotlib < 3.1
     from matplotlib.scale import get_scale_docs
+try:
+    from matplotlib import _docstring
+except ImportError:  # matplotlib < 3.6
+    try:
+        from matplotlib import docstring as _docstring
+    except ImportError:  # pragma: no cover
+        # maybe matplotlib >= 3.9?
+        _docstring = None
 
 from astropy import units
 
@@ -497,7 +505,11 @@ for _unit in TIME_UNITS:
     register_gps_scale(_gps_scale_factory(_unit))
 
 # update the docstring for matplotlib scale methods
-docstring.interpd.update(
-    scale=' | '.join([repr(x) for x in get_scale_names()]),
-    scale_docs=get_scale_docs().rstrip(),
-)
+try:
+    _docstring.interpd.update(
+        scale=' | '.join([repr(x) for x in get_scale_names()]),
+        scale_docs=get_scale_docs().rstrip(),
+    )
+except AttributeError:  # pragma: no cover
+    # matplotlib._docstring changed/removed/not found
+    pass

